pkg/bff/gds.go

```diff
@@ -32,3 +41,95 @@ func ConnectGDS(conf config.DirectoryConfig) (_ gds.TRISADirectoryClient, err er
 	}
 	return gds.NewTRISADirectoryClient(cc), nil
 }
+
+// Lookup makes a request on behalf of the user to both the TestNet and MainNet GDS
+// servers, returning 1..2 results (e.g. either or both GDS responses). If no results
+// are returned, Lookup returns a 404 not found error. If one of the GDS requests fails,
+// the error is logged, but the valid response is returned. If both GDS requests fail,
+// a 500 error is returned. This endpoint passes through the response from GDS as JSON,
+// the result should contain a registered_directory field that identifies which network
+// the record is associated with.
+func (s *Server) Lookup(c *gin.Context) {
+	// Bind the parameters associated with the lookup
+	params := &api.LookupParams{}
+	if err := c.ShouldBindQuery(&params); err != nil {
+		log.Warn().Err(err).Msg("could not bind request with query params")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(err))
+		return
+	}
+
+	// Ensure that we have either ID or CommonName
+	if params.ID == "" && params.CommonName == "" {
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("must provide either uuid or common_name in query params"))
+		return
+	}
+
+	// Create the LookupRequest, omit registered directory as it is assumed we're
+	// looking up the value for the directory we're making the request to.
+	req := &gds.LookupRequest{Id: params.ID, CommonName: params.CommonName}
+	errs := make([]error, 2)
+	results := make([]map[string]interface{}, 2)
+
+	// Execute the request in parallel to both the testnet and the mainnet
+	ctx, cancel := context.WithTimeout(c.Request.Context(), 25*time.Second)
+	var wg sync.WaitGroup
+	defer cancel()
+	wg.Add(2)
+
+	lookup := func(client gds.TRISADirectoryClient, idx int, name string) {
+		defer wg.Done()
+		var (
+			err error
+			rep *gds.LookupReply
+		)
+
+		if rep, err = client.Lookup(ctx, req); err != nil {
+			serr, _ := status.FromError(err)
+			if serr.Code() != codes.NotFound {
+				log.Error().Err(err).Str("network", name).Msg("GDS lookup unsuccessful")
+				errs[idx] = err
+			}
+			return
+		}
+
+		if results[idx], err = wire.Rewire(rep); err != nil {
+			log.Error().Err(err).Str("network", name).Msg("could not rewire LookupReply")
+			errs[idx] = err
+		}
+	}
+
+	go lookup(s.testnet, 0, "testnet")
+	go lookup(s.mainnet, 1, "mainnet")
+	wg.Wait()
```

Comment:
The Register endpoint will query either one or the other since the data will be different and the registration will likely occur at different times - so the use case of querying one network is definitely a thing.

I'll create a story for creating the generic solution for this.